src/variable.js

```diff
@@ -40,10 +40,20 @@ function replaceListDelimiters(varValue, varName = '') {
  * @returns {String} Converted value
  */
 function resolveEnvVars(varValue) {
-  const envUnixRegex = /\$(\w+)|\${(\w+)}/g // $my_var or ${my_var}
-  return varValue.replace(envUnixRegex, (_, varName, altVarName) => {
-    return process.env[varName || altVarName] || ''
-  })
+  const envUnixRegex = /(\\*)(\$(\w+)|\${(\w+)})/g // $my_var or ${my_var} or \$my_var
+  return varValue.replace(
+    envUnixRegex,
+    (_, escapeChars, varNameWithDollarSign, varName, altVarName) => {
+      // do not replace things preceded by a odd number of \
+      if (escapeChars.length % 2 === 1) {
+        return varNameWithDollarSign
+      }
+      return (
+        escapeChars.substr(0, escapeChars.length / 2) +
```

Comment:
Could you walk me through why only half the back-slashes are required? I'm not sure I understand.

Reply:
That's what happens in regular js string escaping too. `\\` gets to be a `\`. This is needed to allow somebody to write a `\` in front of a `$VARIABLE` and still replace the variable.

`\$VAR` -> `$VAR`
`\\$VAR` -> `\value`
`\\\$VAR` -> `\\$VAR`
`\\\\$VAR` -> `\\value`

and so on.

Hope this makes it clear

Reply:
OK. Thank you. What about the other case? If the number of back-slashes is odd, isn't this code going to replace the expression by `$VAR` every time? Shouldn't some back-slashes be retained?

Also I'm not sure your examples are quite consistent. In `\$VAR` -> `$VAR`, the back-slash is escaping the dollar sign. In `\\$VAR` -> `\value` the first back-slash escapes the second and the dollar sign is not escaped. Sounds fine up to there. Then in `\\\$VAR` -> `\\$VAR`, the first black-slash escapes the second, then the third escapes the dollar sign. So the result should be `\$VAR`, shouldn't it? Maybe you made a typo. Fourth example looks right.